runners/spark/src/main/java/org/apache/beam/runners/spark/translation/SparkPCollectionView.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.beam.runners.spark.translation;
+
+import java.io.Serializable;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import org.apache.beam.runners.spark.coders.CoderHelpers;
+import org.apache.beam.runners.spark.util.BroadcastHelper;
+import org.apache.beam.sdk.coders.Coder;
+import org.apache.beam.sdk.util.WindowedValue;
+import org.apache.beam.sdk.values.PCollectionView;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.spark.api.java.JavaSparkContext;
+
+public class SparkPCollectionView implements Serializable {
```

Comment:
This should probably host some kind of thread-safe, lazy, Singleton. this might also save the trouble of passing this through the `TransformEvaluator`.